### PR TITLE
Fixed and added support for dataUri in form data

### LINKED ISSUE #25790: [Android] FormData does not support data URIs
React Native version: Expo 33 / React Native 0.59.8

## Steps To Reproduce

1. Call `formData.append` with a data URI as the `uri`
2. Make XHR request and send FormData

```js
const dataUri = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQYV2NgYAAAAAMAAWgmWQ0AAAAASUVORK5CYII=';
const formData = new FormData();
formData.append('file', {
  name: 'myfile',
  type: 'image/png',
  uri: dataUri,
});

const request = new XMLHttpRequest();
request.open('POST', 'http://httpbin.org/post');
request.onreadystatechange = function() {
  if (request.readyState === 4) {
    if (request.status === 200) {
      console.log("SUCCESS")
      // will succeed on iOS
    } else {
      console.log(request.responseText);
      // response on Android will be "Could not retrieve file for uri data:image/png;base64,iVBO..."
    }
  }
};
request.send(formData);
```

Describe what you expected to happen:

The request would succeed (as it does on iOS) but instead (on Android only), you get this error:

>Could not retrieve file for uri data:image/png;base64,iVB...

Snack: https://snack.expo.io/@maxholderc5/cd0ef5

It appears that the [Android implementation](https://github.com/facebook/react-native/blob/cf77067f0c4eda30f814aafa45ddf962eadcfbdb/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L700-L709) assumes this will always be a file, but unless I've misunderstood, passing a data URI here should also be valid (at least, it seems to work on iOS).

A workaround for my specific scenario (in case it is helpful for others running into this) was to write a temporary file and provide its filepath instead of the data URI directly, e.g. with Expo:

```js
const tempFileDirectory = `${FileSystem.cacheDirectory}uploadTmp`
await FileSystem.makeDirectoryAsync(tempFileDirectory, { intermediates: true })
const tempFilePath = `${tempFileDirectory}${Date.now()}.png`
const base64EncodedData = dataUri.split(',')[1]
await FileSystem.writeAsStringAsync(tempFilePath, base64EncodedData, {
  encoding: FileSystem.EncodingType.Base64,
})
// ...
formData.append(name, {
  name: "myfile",
  type: "image/png",
  uri: tempFilePath,
})
// ...
await FileSystem.deleteAsync(tempFilePath)
```

### PR DESCRIPTION
**Pull request by hetanthakkar1**
## Summary

Continuation of #33548
 Fixes #25790. The issue resulted because the [getFileInputStream() method of RequestBodyUtil failed to return an inputStream of the given dataUri ](https://github.com/facebook/react-native/blob/16397e0d3c0dd3374ddb642599725ca41f092a8a/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L427) This happened because the openInputStream() method of [context.getContentResolver() expects a content or a file](https://developer.android.com/reference/android/content/ContentResolver#openInputStream(android.net.Uri)) in its parameter but instead received dataUri which resulted in FileNotFoundException.

issue 

**Solution**:

I've now converted the dataUri to bitmap and then attached an inputStream to the compressed bitmap. This way we won't have to store the image temporarily. I think converting the dataUri to Bitmap is necessary as there is no method that lets us convert the dataUri to inputStream directly. And regarding large size images, converting them to bitmap is the only(efficient) step I can think of, as the conversion is handled by in-built java function.

This issue has been unresolved for quite some time now, so resolving this PR would be greatly appreciated.
## Changelog

[Android] [Added] - Support for dataUri in form data

### CODE REVIEW

**Comment by hetanthakkar1** [comment]
@cortinico I've removed that unnecessary line.

**Comment by hetanthakkar1** [comment]
@cortinico Can you please review this PR

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java [hunk 1]
```diff
@@ -58,6 +62,16 @@ public static boolean isGzipEncoding(@Nullable final String encodingType) {
       if (fileContentUri.getScheme().startsWith("http")) {
         return getDownloadFileInputStream(context, fileContentUri);
       }
+
+      if (fileContentUriStr.startsWith("data:")) {
+        byte[] decodedDataUrString = Base64.decode(fileContentUriStr.split(",")[1], Base64.DEFAULT);
+        Bitmap bitMap = BitmapFactory.decodeByteArray(decodedDataUrString, 0, decodedDataUrString.length);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bitMap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
```
Comment (cortinico) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: This won't work as it as you're compressing with JPG an image that could be potentially a PNG/GIF or other formats.

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: @cortinico compressFormat also supports webp and PNG! We can add them too!?

Comment (cortinico) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: It seems like the correct approach would be:
```
bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
```

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: @cortinico Do you mean something like this?
if(format is PNG)
bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
if(format is JPG)
bitMap.compress(CompressFormat.JPG, 0 /* ignored for JPG */, bytes);
if(format is WEBP)
bitMap.compress(CompressFormat.WEBP, 0 /* ignored for WEBP */, bytes);

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: > It seems like the correct approach would be:
> 
> ```
> bitMap.compress(CompressFormat.PNG, 0 /* ignored for PNG */, bytes);
> ```

Or should we do this just for PNG format, Sorry I didn’t completely get your approach

Comment (cortinico) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:70]: The snippet I suggested essentially ignores compression, so you don't need to check the format at all

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java]: @cortinico I've made the changes. does the changes looks good?

**Comment by hetanthakkar1** [comment]
A test is failing, I don’t think the failure is related to my changes!? Is there anything else I need to do

**Review by cortinico — APPROVED** [review]

**Comment by hetanthakkar1** [comment]
@cortinico Internal tests are failing. Do i need to do make any corrections!? Or it's nothing to worry about

**Comment by cortinico** [comment]
Nope I'll take care of it next week 👍

**Comment by hetanthakkar1** [comment]
@cortinico Okay Thank you

**Pull request closed without merging**